lib/model.js

```diff
@@ -4214,10 +4214,14 @@ Model.aggregate = function aggregate(pipeline, options, callback) {
 
   const aggregate = new Aggregate(pipeline || []);
   aggregate.model(this);
-
   if (options != null) {
     aggregate.option(options);
   }
+  const defaultAllowDiskUse = this.base.options.allowDiskUse || false;
+  const operationAllowDiskUse = options && options.allowDiskUse;
+  if (operationAllowDiskUse == null && defaultAllowDiskUse != null) {
```

Comment:
Can defaultAllowDiskUse ever be undefined or null? In line 4220 it will always be set to false if allowDiskUse is null or undefined. So I would say the second condition is always truthy. So we should maybe just set allowDiskUse if we didn't set allowDiskUse on the operation and allowDiskUse is set globally.

```suggestion
  if (operationAllowDiskUse == null && defaultAllowDiskUse === true) {
```

Or simpler

```suggestion
  if (operationAllowDiskUse == null && defaultAllowDiskUse) {
```

Reply:
Good catch, I changed the logic a little bit to assign the default value as long as `operationAllowDiskUse == null`.
Ther reason I'm trying to avoid `defaultAllowDiskUse === true` is because using whatever the user sends to the driver is a good idea, for example, if a user sends `'true'`/`1`/`0`, we'd wanna send it to the driver instead of not sending anything at all.